service/reportsStorage/handlers.go

```diff
@@ -0,0 +1,188 @@
+package reportsStorage
+
+import (
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/rs/zerolog"
+
+	"github.com/iryonetwork/wwm/gen/reports/models"
+	"github.com/iryonetwork/wwm/gen/reports/restapi/operations"
+	"github.com/iryonetwork/wwm/utils"
+)
+
+// Handlers describes the actions supported by the storage handlers
+type Handlers interface {
+	ReportList() operations.ReportListHandler
+	ReportGet() operations.ReportGetHandler
+	ReportGetVersion() operations.ReportGetVersionHandler
+	ReportListVersions() operations.ReportListVersionsHandler
+	ReportNew() operations.ReportNewHandler
+	ReportUpdate() operations.ReportUpdateHandler
+	ReportDelete() operations.ReportDeleteHandler
+}
+
+type handlers struct {
+	service Service
+	logger  zerolog.Logger
+}
+
+func (h *handlers) ReportList() operations.ReportListHandler {
+	return operations.ReportListHandlerFunc(func(params operations.ReportListParams, principal *string) middleware.Responder {
+		list, err := h.service.ReportList(params.HTTPRequest.Context(), params.ReportType)
+
+		if err != nil {
+			return operations.NewReportListInternalServerError().WithPayload(&models.Error{
+				Code:    "server_error",
+				Message: err.Error(),
+			})
+		}
+		if len(list) == 0 {
+			return operations.NewReportListNotFound()
+		}
+
+		return operations.NewReportListOK().WithPayload(list)
+	})
+}
+
+func (h *handlers) ReportGet() operations.ReportGetHandler {
+	return operations.ReportGetHandlerFunc(func(params operations.ReportGetParams, principal *string) middleware.Responder {
+		r, fd, err := h.service.ReportGet(params.HTTPRequest.Context(), params.ReportType, params.FileName)
+
+		if err != nil {
+			switch err {
+			case ErrNotFound:
+				return operations.NewReportGetNotFound()
+			default:
+				return operations.NewReportGetInternalServerError().WithPayload(&models.Error{
+					Code:    "server_error",
+					Message: err.Error(),
+				})
+			}
+		}
+
+		return utils.UseProducer(operations.NewReportGetOK().
+			WithPayload(r).
+			WithXContentType(fd.ContentType).
```

Comment:
invalid operation: fd (variable of type *invalid type) has no field or method ContentType